### DISCUSSION TITLE
benefits of wgrib2 on linux

### DISCUSSION THREAD

**Discussion by roesgaard** [blaylockbk/Herbie#279] [Q&A]
In the conda environment.yml there is an optional install for linux for wgrib2.

What are the benefits of using this? faster/easier data handling?
and will Herbie start using it by default or should I change the code accordingly?

Thanks in advance!

**Comment by blaylockbk** [comment]
Wgrib2 can do two things that are useful to Herbie

1. Generate an index of the GRIB2 file contents. This is useful for model data that don't provide an index file. The index file is used by Herbie to download specific GRIB2 messages.
1. Wgrib2 can create regional subsets from the GRIB2 data (file must be downloaded locally) 

The first use is implemented by Herbie; for a full grib2 file that is downloaded locally, and an index file is not available, Herbie will generate one with wgrib2 if wgrib2 is installed.

The second use is only half baked in Herbie.

Essentially, using wgrib2 can be convenient for some things that are too hard to do with raw python. It is possible that the eccodes library can do some of these things wgrib2 can do, but I haven't investigated.
